internal/resource-request-operator/resourceRequest_controller.go

```diff
@@ -47,6 +49,17 @@ func (r *ResourceRequestReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	remoteClusterID := resourceRequest.Spec.ClusterIdentity.ClusterID
 
+	if resourceRequest.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&resourceRequest, cleanBroadcasterFinalizer) {
+			controllerutil.AddFinalizer(&resourceRequest, cleanBroadcasterFinalizer)
+		}
+	} else {
+		if controllerutil.ContainsFinalizer(&resourceRequest, cleanBroadcasterFinalizer) {
+			r.Broadcaster.RemoveClusterID(remoteClusterID)
+			controllerutil.RemoveFinalizer(&resourceRequest, cleanBroadcasterFinalizer)
+		}
+	}
```

Comment:
Wondering whether the finalizer could be completely dropped. Theoretically, it should be sufficient to remove the clusterID if the get returns a notfound error at the beginning, avoiding the finalizer complexity. @palexster WDYT?

Reply:
100%